pkgs/development/libraries/dbus/default.nix

```diff
@@ -98,6 +102,11 @@ stdenv.mkDerivation rec {
 
   doCheck = true;
 
+  makeFlags = [
+    # Fix paths in XML catalog broken by mismatching build/install datadir.
+    "dtddir=${placeholder "out"}/share/xml/dbus-1"
```

Comment:
Alternative might be a more targetted approach by overriding `"dbusdatadir=${placeholder "out"}/share/dbus-1"` in `installFlags` and `"dbusdatadir=${placeholder "out"}/etc/dbus-1"` in `makeFlags`, instead of setting custom `datadir`.

Ideally, we would just drop the custom `datadir` altogether. Unfortunately, it caused issues in the past and got reverted: https://github.com/NixOS/nixpkgs/commit/b3b123a104f448e1f377b162cbd52af461fdecb2